pallets/handles/src/handles_signed_extension.rs

```diff
@@ -0,0 +1,133 @@
+//! Substrate Signed Extension for validating requests to the handles pallet
+use crate::{Call, Config, Error, MSAIdToDisplayName};
+use codec::{Decode, Encode};
+use common_primitives::msa::MsaValidator;
+use core::marker::PhantomData;
+use frame_support::{
+	dispatch::{DispatchInfo, Dispatchable},
+	ensure,
+	pallet_prelude::ValidTransaction,
+	traits::IsSubType,
+	unsigned::UnknownTransaction,
+};
+use scale_info::TypeInfo;
+use sp_runtime::{
+	traits::{DispatchInfoOf, SignedExtension},
+	transaction_validity::{InvalidTransaction, TransactionValidity, TransactionValidityError},
+	DispatchError, ModuleError,
+};
+/// The SignedExtension trait is implemented on CheckFreeExtrinsicUse to validate the request. The
+/// purpose of this is to ensure that the retire_handle extrinsic cannot be
+/// repeatedly called to flood the network.
+#[derive(Encode, Decode, Clone, Eq, PartialEq, TypeInfo)]
+#[scale_info(skip_type_params(T))]
+pub struct HandlesSignedExtension<T: Config + Send + Sync>(PhantomData<T>);
+
+impl<T: Config + Send + Sync> HandlesSignedExtension<T> {
+	/// Create new `SignedExtension`.
+	pub fn new() -> Self {
+		Self(sp_std::marker::PhantomData)
+	}
+
+	/// Validates the following criteria for the retire_handle() extrinsic:
+	///
+	/// * The delegator must already have a MSA id
+	/// * The MSA must already have a handle associated with it
+	///
+	/// Returns a `ValidTransaction` or wrapped [`pallet::Error`]
+	///
+	/// # Errors (as u8 wrapped by `InvalidTransaction::Custom`)
+	/// * [`Error::InvalidMessageSourceAccount`]
+	/// * [`Error::MSAHandleDoesNotExist`]
+	fn validate_retire_handle(delegator_key: &T::AccountId) -> TransactionValidity {
+		const TAG_PREFIX: &str = "HandlesRetireHandle";
+
+		// Validation: The delegator must already have a MSA id
+		let delegator_msa_id = T::MsaInfoProvider::ensure_valid_msa_key(&delegator_key)
+			.map_err(|e| map_dispatch_error(e))?;
+		// Validation: The MSA must already have a handle associated with it
+		let handle_from_state = MSAIdToDisplayName::<T>::try_get(delegator_msa_id)
+			.map_err(|_| UnknownTransaction::CannotLookup)?;
+		let expiration = handle_from_state.1;
+		let current_block = frame_system::Pallet::<T>::block_number();
+
+		let is_past_min_lifetime = current_block >= expiration;
+		ensure!(
+			is_past_min_lifetime,
+			map_dispatch_error(DispatchError::Other(
```

Comment:
I do not think this is doing anything. It just gets unknown error? I am not sure what your intent is here.

Reply:
I think there is an error added in Other::, basically it prevents expiration if, current block hasnt passed the blocker number set with handle (mortality check)

Reply:
`			map_dispatch_error(DispatchError::Other(
				Error::<T>::HandleWithinMortalityPeriod.into()
			))`